### ISSUE TITLE
Subdirectory with the same name as the parent doesn't appear in listdir()

### ISSUE THREAD

**Issue by samuelsinayoko** [PyFilesystem/s3fs#3]
Given a directory structure
```
bucket/dir/dir
```
doing
```
from fs import open_fs
s3fs = open_fs('s3://bucket')
s3fs.listdir('dir')
```
returns the empty list. I reckon this is because of line 440 in `_s3fs.py`:
```
~/miniconda3/envs/myenv/lib/python3.4/site-packages/fs_s3fs/_s3fs.py in listdir(self, path)
    438                     _prefix = prefix.get('Prefix')
    439                     _name = _prefix[prefix_len:]
--> 440                     if _name != _s3_key:
    441                         _directory.append(_name.rstrip(self.delimiter))
    442                 for obj in result.get('Contents', ()):
```
In this example, `_s3_key == _name == 'dir'` so the subdirectory isn't included in the list of directories :grimacing:

**Comment by samuelsinayoko** [comment]
It seems that replacing line 440 by `if prefix_len:` resolves the issue with `listdir`. 

This same line `if _name != _s3_key` seems to break `S3FS.geninfo` but again replacing by `if prefix_len` seems to solve the issue.

**Comment by samuelsinayoko** [comment]
See PR here describing the proposed fixes https://github.com/bmlltech/s3fs/pull/1/files

**Comment by willmcgugan** [comment]
Thanks. Will look at that soon.

**Comment by willmcgugan** [comment]
Should be fixed in 0.1.3
